src/main/java/core/db/DBUpdater.java

```diff
@@ -116,13 +116,17 @@ void updateDB(int DBVersion) {
 		}
 	}
 
-	private void updateDBv400(int dbVersion, int version){
+	private void updateDBv400(int dbVersion, int version) throws SQLException {
```

Comment:
I am also in favour of consistency.  In this particular case, if the exception is handled outside this method, this might make it tricky to have a recovery strategy?

This does beg the question, what should be our strategy in case of an exception in the db updater?  Should we make an attempt at rollback?

Reply:
the strategy should be
- catch it only if we have an idea how to handle the exception to get a valid updated database
- outside of the special update routines is an catch that simply logs the exception. if the inner catch would do nothing else it should be omitted. The database version info is not updated.

Reply:
@wsbrenk but the database will be in a partially updated state ...  

> Should we make an attempt at rollback?

either that but it might be difficult or we try to restore the database that was backed up on startup with a clear message to the user of what just happened ...

Reply:
@akasolace we should distinguish two things here
- try catch strategy should be: catch only it we have a chance to get a correct updated database
- transaction strategy that handles a rollback of invalid partly changes of the database (e. g. restore a backup)

Reply:
@wsbrenk I agree. This is not directly link to this specific commit but we need to have a common approach for this. I mean which process if dbupdater raise an exception. This should be a ticket we need to discuss at some point ...

Reply:
ok I created a ticket for this